File: Semestrul 4/MPP2/Problema8_FX_Csharp/Problema8_FX_Csharp/Domain/Event.cs
namespace Problema8_FX_Csharp.Domain;

public class Event : Entity<long>
{
    public String Name { get; set; }
    public int Distance { get; set; }
    
    public Event(long id, String name, int distance)
    {
        SetId(id);
        Name = name;
        Distance = distance;
    }
    
    public override String ToString()
    {
        return $"Name: {Name}, Distance: {Distance}";
    }
    
    public override bool Equals(Object obj)
    {
        if (obj == null || GetType() != obj.GetType())
        {
            return false;
        }
        
        Event e = (Event)obj;
        return Name.Equals(e.Name) && Distance == e.Distance;
    }
}

File: Semestrul 4/MPP2/Problema8_SC/Problema8_SC/Problema8Model/src/main/java/problema8/model/Event.java
package problema8.model;

import java.util.Objects;

public class Event extends Entity<Long> {
    private String name;
    private Integer distance;
    private Long ageGroupId;

    public Event(Long id, String name, Integer distance, Long ageGroupId) {
        setId(id);
        this.name = name;
        this.distance = distance;
        this.ageGroupId = ageGroupId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getDistance() {
        return distance;
    }

    public void setDistance(Integer distance) {
        this.distance = distance;
    }

    public Long getAgeGroupId() {
        return ageGroupId;
    }

    public void setAgeGroupId(Long ageGroupId) {
        this.ageGroupId = ageGroupId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Event event = (Event) o;
        return Objects.equals(name, event.name) && Objects.equals(distance, event.distance) && Objects.equals(ageGroupId, event.ageGroupId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, distance, ageGroupId);
    }

    @Override
    public String toString() {
        return "Event{" +
                "name='" + name + '\'' +
                ", distance=" + distance +
                ", ageGroupId=" + ageGroupId +
                '}';
    }
}
